packages/metro-config/src/defaults/__tests__/exclusionList-test.js

```diff
@@ -25,14 +25,84 @@ describe('exclusionList', () => {
     // $FlowFixMe: property sep is not writable.
     path.sep = '/';
+    expect(require('path').sep).toBe('/');
+    // $FlowFixMe: property sep is not writable.
+    path.sep = '\\';
+    expect(require('path').sep).toBe('\\');
+  });
+
+  it('converts forward slashes in the RegExp to the OS specific path separator in macOS/linux', () => {
+    // $FlowFixMe: property sep is not writable.
+    path.sep = '/';
+    // Simple case
     expect('a/b/c').toMatch(exclusionList([new RegExp('a/b/c')]));
+    expect('a/b/c').toMatch(exclusionList([/a\/b\/c/]));
+    // Regular expression that already considered OS specific path separator.
+    // When explictly construct RegExp instance, the string needs to be escaped first.
+    // eg. /.*[/\\]foo[/\\]bar/ => '.*[/\\\\]foo[/\\\\]bar'
+    expect('/foo/bar').toMatch(exclusionList([new RegExp('.*[/\\\\]foo[/\\\\]bar')]));
+    expect('/foo/bar').toMatch(exclusionList([/.*[/\\]foo[/\\]bar/]));
+  });
 
+  it('converts forward slashes in the RegExp to the OS specific path separator in windows', () => {
     // $FlowFixMe: property sep is not writable.
     path.sep = '\\';
-    expect(require('path').sep).toBe('\\');
+    // Simple case
     expect('a\\b\\c').toMatch(exclusionList([new RegExp('a/b/c')]));
+    expect('a\\b\\c').toMatch(exclusionList([/a\/b\/c/]));
+    // Regular expression that already considered OS specific path separator.
+    // When explictly construct RegExp instance, the string needs to be escaped first.
+    // eg. /.*[/\\]foo[/\\]bar/ => '.*[/\\\\]foo[/\\\\]bar'
+    expect('\\foo\\bar').toMatch(exclusionList([new RegExp('.*[/\\\\]foo[/\\\\]bar')]));
+    expect('\\foo\\bar').toMatch(exclusionList([/.*[/\\]foo[/\\]bar/]));
+  });
+
+  it('converts forward slashes in the string to the OS specific path separator in macOS/linux', () => {
+    // $FlowFixMe: property sep is not writable.
+    path.sep = '/';
+    // Simple case
+    expect('a/b/c').toMatch(exclusionList(['a/b/c']));
+    // Strings that already considered OS specific path separator.
+    expect('/foo/bar').toMatch(exclusionList(['.*[/\\]foo[/\\]bar']));
+  });
+
+  it('converts forward slashes in the string to the OS specific path separator in windows', () => {
+    // $FlowFixMe: property sep is not writable.
+    path.sep = '\\';
+    // Simple case
+    expect('a\\b\\c').toMatch(exclusionList(['a/b/c']));
+    // Strings that already considered OS specific path separator.
+    expect('\\foo\\bar').toMatch(exclusionList(['.*[/\\]foo[/\\]bar']));
+  });
+
+  it('converts forward slashes in the RegExp to the OS specific path separator for macOS/linux in nodejs 10 or below', () => {
+    // In node version 10 or below, the forward slash in brackets are escaped automatically.
+    // eg. /[/\\]/ => /[\/\\]/
+    // Ideally this test case should be removed and instead the whole test should run in
+    // multiple node versions.
+    // $FlowFixMe: property sep is not writable.
```

Comment:
I believe we can emulate the node 10 behavior by writing a single test that asserts that two RegExp are the same regardless if the slash is automatically escaped.

```
const escaped = exclusionList(/[\/]/);
const nonEscaped = exclusionList(/[/]/); 

expect(nonEscaped.source).toEqual(escaped.source);

Reply:
I like this! I will add to the test list.

Reply:
Give it another thought. They are actually not the same. 
```
const escaped = exclusionList(/[\/]/); // [\\\\]
const nonEscaped = exclusionList(/[/]/); // [\\]
```

However, for regular expression, `/[\\]/` is the same to `/[\\\\]/` as we are just repeating the options in the square brackets.